src/activatable/activatable.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Created ->- Activating ->- Activated ->- Deactivating ->- Deactivated -------------->----------- Disposed
+ *                 |  |                                         |  |                                  |  |
+ *                 |  `------------------<----------------------'  `--->--- FailedToDeactivate --->---'  |
+ *                 `--------------------->--------- FailedToActivate ------------------>-----------------'
+ */
+
+enum LifeCycleState {
```

Comment:
```suggestion
enum LifecycleState {
```
I think `Lifecycle` is more common?

Reply:
I guess it's too late to hyphenate the package name.

Reply:
oh wait I thought this was backwards. Um... is it?

Reply:
> Um... is it?

I think so. At least React uses a single word 'lifecycle' everywhere. I see there are some alternatives with the same meaning, such as 'life cycle', but React team uses 'lifecycle', so maybe let's follow the convention?

Reply:
good ol' [stack exchange](https://english.stackexchange.com/a/68233). This graph is awesome:
![dR55h](https://user-images.githubusercontent.com/587740/64976427-3d282c80-d87f-11e9-9de4-8edfbe87b890.png)

It looks like `lifecycle` is 📈 and `life cycle` is 📉 so yeah, let's make it one word. Very zeitgeist of us.